### ISSUE TITLE
[BUG] container running inside a vm are performing slow network

### ISSUE THREAD

**Issue by hsinhoyeh** [harvester/harvester#3822]
**Describe the bug**

We were running with a single node harvester installation with version [1.1.2-rc9](https://github.com/harvester/harvester/releases/tag/v1.1.2-rc9). 
from the harvester dashboard, we created a VM with an Ubuntu 20.04 image (called dev01).
what we are trying to do is to install docker on dev01 and ideally to be able to run our container applications on it.

here is our setup in topology:

|-----------------|     |---------------|       |-----------|
| harvester node  | -> | VM1 (dev01) |    -> | container | 
|-----------------|     |---------------|       |-----------|

However, we are experiencing slow network connectivity from any running container on dev01.
we started with the ping test, the result seems okay
```
ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
64 bytes from 8.8.8.8: icmp_seq=1 ttl=113 time=1.93 ms
--- 8.8.8.8 ping statistics ---
6 packets transmitted, 6 received, 0% packet loss, time 5009ms
rtt min/avg/max/mdev = 1.926/3.017/5.515/1.204 ms
```
but when run apt-get, the connection is always stuck on "waiting for headers"

```
apt-get update
0% [Waiting for headers] [Waiting for headers]

Err:1 http://archive.ubuntu.com/ubuntu focal InRelease
  Connection failed [IP: 185.125.190.36 80]
Err:2 http://security.ubuntu.com/ubuntu focal-security InRelease
  Connection failed [IP: 185.125.190.36 80]
Err:3 http://archive.ubuntu.com/ubuntu focal-updates InRelease
  Connection failed [IP: 185.125.190.39 80]
```

** Here is our workaround **

we found that NIC configuration (as shown below),  docker0 (bridge network for docker container) is running with mtu=1500 but the actual NIC (enp1s0) is running with mtu=1450. so we guess this would be an issue.

<img width="736" alt="Screenshot 2023-04-25 at 1 46 40 AM" src="https://user-images.githubusercontent.com/2792682/234076031-42c36dba-ac5a-4995-a552-80559675e852.png">

so we forced docker daemon to use mtu 1450 (in /etc/docker/daemon.json) with the following changes:

```
{
        "mtu": 1450
}
```
and restart dockerd with `systemctl restart docker`

the `apt-get` is working as expected.

**To Reproduce**
1. Create a VM
2. install [Docker engine](https://docs.docker.com/engine/install/ubuntu/) on the VM  
3. run `docker run -it ubuntu:20.04 /bin/sh -c "apt-get update"`

**Expected behavior**

`apt-get update` executed on step 3 should work.

**Environment**
 - Harvester ISO version: 1.1.2-rc9
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): not related

**Additional context**

This may not be what the harvester team can do here, but I think it is good to keep it here for reference.

**Comment by starbops** [comment]
Hi @hsinhoyeh, I believe what you did is the right move. The MTU for `dockerd` is 1500 by default. It does not automatically detect the underlying network interfaces for an appropriate MTU value and needs to be configured accordingly by the user.

Did you create the VM `dev01` with a NIC associated with the `management network`?
<img width="1104" alt="Screen Shot 2023-04-26 at 16 00 13" src="https://user-images.githubusercontent.com/1827717/234509651-e9c12c01-fd2c-4d18-bc42-0e2d8aaa72c8.png">

If yes, then `enp1s0`'s MTU value is 1450 because it's on top of the VXLAN network infra. There is some overhead for the tunneling. To make the default 1500 MTU of dockerd work, you can instead create the VM with other [VLAN networks](https://docs.harvesterhci.io/v1.1/networking/harvester-network#vlan-network) to avoid the tunneling network. And you will get a default 1500 MTU for `enp1s0`.

**Comment by hsinhoyeh** [comment]
@starbops thanks for the info. Ya, we did use the default network (because we keep hit with `Dhcp failed` when we tried to create a new VLAN, but that is a separate issue).

We totally understand there is some overhead, but it is not easy to spot the difference on MTU that is the root cause for the "slow network" (as the behavior looks like a slow network issue.).

maybe it is worth some documentation in the faq section?
